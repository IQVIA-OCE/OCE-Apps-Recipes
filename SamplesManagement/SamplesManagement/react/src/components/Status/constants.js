export const statusColors = {
  InProgress: {
    color: '#ffb75d',
  },
  Urgent: {
    color: '#ff6565',
  },
  Submitted: {
    color: '#04844b',
  },
  Approved: {
    color: '#04844b',
  },
  Saved: {
    color: '#5eb4ff',
  },
  Rejected: {
    color: '#c23934',
  },
  Shipped: {
    color: '#04844b',
  },
  BackOrder: {
    color: '#5eb4ff',
  },
  SentToFulfillment: {
    color: '#04844b',
  },
  PendingApproval: {
    color: '#5eb4ff',
  },
  SubmittedforApproval: {
    color: '#5eb4ff',
  },
  Returned: {
    color: '#c23934',
  },
  Duplicate: {
    color: '#5eb4ff',
  },
  Info: {
    color: '#ecebea',
    textColor: '#000'
  }
};
