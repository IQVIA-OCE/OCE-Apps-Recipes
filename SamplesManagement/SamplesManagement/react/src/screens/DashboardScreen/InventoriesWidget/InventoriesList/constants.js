export const inventoriesColors = {
  AdHocInventory: '#b781d3',
  AuditedInventory: '#4ad463',
  InitialInventory: '#769ed9',
  PeriodicInventory: '#ef6e64',
}
