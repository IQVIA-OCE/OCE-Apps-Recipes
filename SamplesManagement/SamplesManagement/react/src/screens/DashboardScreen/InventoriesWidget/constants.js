export const inventoriesColors = {
  AdHocInventory: '#b781d3',
  AuditedInventory: '#4ad463',
  InitialInventory: '#769ed9',
  PeriodicInventory: '#ef6e64',
}

export const INVENTORY_TYPE = {
  InitialInventory: 'Initial Inventory',
  AdHocInventory: 'Ad Hoc Inventory',
  AuditedInventory: 'Audited Inventory',
  PeriodicInventory: 'Periodic Inventory',
};