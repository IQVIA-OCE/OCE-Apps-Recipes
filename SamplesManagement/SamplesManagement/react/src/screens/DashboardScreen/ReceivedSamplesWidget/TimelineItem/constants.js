export const recordTypes = {
  AcknowledgementOfShipment: {
    icon: 'format-list-checks',
    color: '#ef6e64',
  },
  TransferIn: {
    icon: 'format-indent-increase',
    color: '#f88962',
  },
};
