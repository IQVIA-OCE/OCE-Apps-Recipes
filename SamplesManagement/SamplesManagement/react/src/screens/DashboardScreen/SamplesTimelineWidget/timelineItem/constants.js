export const recordTypes = {
  AcknowledgementOfShipment: {
    icon: 'format-list-checks',
    color: '#ef6e64',
  },
  Adjustment: {
    icon: 'tune-vertical',
    color: '#47cfd2',
  },
  Disbursement: {
    icon: 'file-document-box-outline',
    color: '#769ed9',
  },
  TransferIn: {
    icon: 'format-indent-increase',
    color: '#f88962',
  },
  TransferOut: {
    icon: 'format-indent-decrease',
    color: '#f88962',
  },
  Order: {
    icon: 'clipboard-check',
    color: '#69cd37',
  },
  Return: {
    icon: 'package-variant-closed',
    color: '#b781d3',
  },
};
