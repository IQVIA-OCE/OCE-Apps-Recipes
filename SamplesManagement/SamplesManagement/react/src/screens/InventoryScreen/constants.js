export const DATE_TIME_FORMAT = 'MMM D, YYYY h:mm A';
export const DATE_TIME_FORMAT_WITH_SECONDS = 'MMM D, YYYY h:mm:ss A';
export const DATE_FORMAT = 'MMM D, YYYY';
export const INVENTORY_STATUS = {
  inProgress: 'In Progress',
  saved: 'Saved',
  submitted: 'Submitted',
};

export const BANNER_CONFIG = {
  inProgress: {
    variant: 'success',
    message: 'In Progress. Saved Successfully',
    visible: true,
    icon: 'checkbox-marked-circle',
  },
  saved: {
    variant: 'warning',
    message: 'Saved. Need reason for discrepancies',
    visible: true,
    icon: 'alert',
  },
  submitted: {
    variant: 'success',
    message: 'Submitted Successfully',
    visible: true,
    icon: 'checkbox-marked-circle',
  },
};

export const REASON_LIST = [
  { label: 'Theft/Loss', id: 'Theft/Loss' },
  { label: 'Damaged', id: 'Damaged' },
  { label: 'Disbursement Error', id: 'Disbursement Error' },
  { label: 'Other', id: 'Other' },
];
