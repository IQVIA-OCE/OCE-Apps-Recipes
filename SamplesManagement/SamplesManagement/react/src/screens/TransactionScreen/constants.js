export const packageConditions = [
  { label: 'Undamaged', id: '1' },
  { label: 'Damaged', id: '2' },
  { label: 'Opened', id: '3' },
];
