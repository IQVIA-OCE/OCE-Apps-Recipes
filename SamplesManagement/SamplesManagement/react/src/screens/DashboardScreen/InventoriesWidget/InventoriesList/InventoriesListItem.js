import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { inventoriesColors } from './constants';
import Status from '../../../../components/Status/Status';
import moment from 'moment';

const InventoriesListItem = ({ item }) => {
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <View
          style={[
            styles.icon,
            { backgroundColor: inventoriesColors[item.DeveloperName] },
          ]}
        >
          <Icon name="cube-outline" size={32} color={'#fff'}/>
        </View>
        <View>
          <Text style={styles.title}>{item.OCE__SampleInventoryDetails__r.totalSize} Product Sample Details</Text>
          <View style={styles.descriptionContainer}>
            <Text style={styles.descriptionTitle}>Date: </Text>
            <Text style={styles.descriptionText}>{moment(item.OCE__InventoryDateTime__c).format('MMM d, YYYY hh:mm a')}</Text>
          </View>
        </View>
      </View>
      <View>
        <Status status={item.OCE__Status__c}/>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    width: 32,
    height: 32,
    borderRadius: 3,
    marginRight: 10,
    alignItems: 'center',
  },
  descriptionContainer: {
    flexDirection: 'row',
    marginTop: 5,
  },
  descriptionTitle: {
    fontSize: 12,
    color: '#3e3e3c',
  },
  descriptionText: {
    fontSize: 12,
    color: '#080707',
  },
  title: {
    color: '#0070d2',
    fontSize: 13,
  },
});

export default InventoriesListItem;
