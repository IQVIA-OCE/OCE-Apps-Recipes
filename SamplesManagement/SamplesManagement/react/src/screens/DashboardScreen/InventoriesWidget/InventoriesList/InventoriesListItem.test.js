import React from 'react';
import InventoriesListItem from './InventoriesListItem';
import { TouchableOpacity } from 'react-native';
import { render, act } from '@testing-library/react-native';
import { useNavigation } from '@react-navigation/native';
import { NAMESPACE } from '../../../../constants/constants';

jest.mock('@react-navigation/native');

const mockedNavigate = jest.fn();

useNavigation.mockImplementation(() => ({
  navigate: mockedNavigate
}));

let item;

describe('InventoriesListItem', () => {
  beforeEach(() => {
    item = {
      [`${NAMESPACE}SampleInventoryDetails__r`]: {
        totalSize: 1,
      },
      [`${NAMESPACE}InventoryDateTime__c`]: '2020-03-13T16:53:50.000Z',
      [`${NAMESPACE}Status__c`]: 'Submitted',
      [`${NAMESPACE}Reason__c`]: 'Test reason'
    };
  });

  it('Should render InventoriesListItem component', () => {
    const { getByText, UNSAFE_root } = render(
      <InventoriesListItem item={item} />
    );

    act(() => {
      UNSAFE_root.findAllByType(TouchableOpacity)[0].props.onPress();
    });

    expect(getByText(/Test reason/)).toBeTruthy();
    expect(mockedNavigate).toBeCalledTimes(1);
  });
});
