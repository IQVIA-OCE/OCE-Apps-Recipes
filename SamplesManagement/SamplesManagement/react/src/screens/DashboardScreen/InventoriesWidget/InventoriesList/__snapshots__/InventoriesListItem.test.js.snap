// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoriesListItem Should render InventoriesListItem component 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 3,
              "height": 32,
              "marginRight": 10,
              "width": 32,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 32,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï†§
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#0070d2",
              "fontSize": 13,
            }
          }
        >
          1 
          Product Sample Details
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3e3e3c",
                "fontSize": 12,
              }
            }
          >
            Date: 
          </Text>
          <Text
            style={
              Object {
                "color": "#080707",
                "fontSize": 12,
              }
            }
          >
            Mar 13, 2020 04:53 PM
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "borderRadius": 10,
              "flexDirection": "row",
              "paddingHorizontal": 6,
              "paddingVertical": 2,
            },
            undefined,
            Object {
              "backgroundColor": "#04844b",
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 12,
                "fontWeight": "700",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Submitted
        </Text>
      </View>
    </View>
  </View>
</View>
`;
